data_filter_mongodb/internal/mongo/mongo.go

```diff
@@ -0,0 +1,144 @@
+package mongo
+
+import (
+	"context"
+	"encoding/json"
+	"log"
+	"net/http"
+
+	"go.uber.org/zap"
+
+	"go.mongodb.org/mongo-driver/bson"
+	"go.mongodb.org/mongo-driver/mongo"
+	"go.mongodb.org/mongo-driver/mongo/options"
+)
+
+type DBClient struct {
+	Address string
+	Client  *mongo.Client
+}
+
+type MongoClientStruct struct {
+	Mongo         *mongo.Client
+	Database      string
+	ClientOptions *options.ClientOptions
+	Logger        *zap.Logger
+}
+
+type MongoClient interface {
+	QueryMongo(pipeline []Queries) Result
+	GetAllDataFromMongo(w http.ResponseWriter)
+}
+
+// Result contains partially evaluating OPA queries.
+type Result struct {
+	Defined bool
+	Data    []EmployeeDetails
+}
+
+type Queries struct {
+	Pipeline bson.M
+}
+
+func (c *MongoClientStruct) CreateConnection() *mongo.Client {
+	client, err := mongo.Connect(context.Background(), c.ClientOptions)
+	if err != nil {
+		c.Logger.Error("Unable to establish connection with mongo client",
+			zap.String("uri", c.ClientOptions.GetURI()), zap.Error(err))
+	}
+
+	err = client.Ping(context.TODO(), nil)
+	if err != nil {
+		c.Logger.Error("Unable to ping the mongo db ", zap.Error(err))
+	}
+	return client
+}
+
+func (c *MongoClientStruct) QueryMongo(pipeline []Queries) Result {
```

Comment:
What happens on an error ?

Reply:
We log the error and return an empty result set.